src/main/sensors/battery.c

```diff
@@ -232,16 +233,48 @@ static void batteryUpdateVoltageState(void)
         default:
             break;
     }
+
+}
+
+static void batteryUpdateLVC(timeUs_t currentTimeUs) 
+{   
+    if (batteryConfig()->lvcEnabled) {
+        if (voltageState == BATTERY_CRITICAL && !lowVoltageCutoff.enabled) {
+            lowVoltageCutoff.enabled = true;
+            lowVoltageCutoff.startTime = currentTimeUs;
+            lowVoltageCutoff.percentage = 100;
+        }
+        if (lowVoltageCutoff.enabled) {
+            if (currentTimeUs - lowVoltageCutoff.startTime < LVC_AFFECT_TIME) {
+                lowVoltageCutoff.percentage = 100 - ((currentTimeUs - lowVoltageCutoff.startTime) * (100 - batteryConfig()->lvcPercentage) / LVC_AFFECT_TIME); 
```

Comment:
lowVoltageCutoff.percentage could be float instead 50% -> .5, Then all that math could be done in float... but not sure if I like all that casting in fc_rc.c.  

```tmp = tmp * getLowVoltageCutoff()->percentage / 100;```

to

```tmp = (uint32)(tmp * getLowVoltageCutoff()->ratio);```